tests/string/BigIntString.test.ts

```diff
@@ -0,0 +1,89 @@
+import * as RA from 'fp-ts/ReadonlyArray'
+import * as E from 'fp-ts/Either'
+import { pipe, tuple } from 'fp-ts/function'
+import * as BigIntString from '../../src/string/BigIntString'
+import { cat, combineExpected, validateArbitrary } from '../../test-utils'
+
+const valid: ReadonlyArray<string> = ['0', '10', '-1', '11']
+
+const invalid: ReadonlyArray<string> = ['5.5', '-5.5', '', ' ', 'a', 'a5']
+
+describe('BigIntString', () => {
+  describe('Decoder', () => {
+    test.each(cat(combineExpected(valid, 'Right'), combineExpected(invalid, 'Left')))(
+      'validates valid strings, and catches bad strings',
+      (str, expectedTag) => {
+        const result = BigIntString.Decoder.decode(str)
+        expect(result._tag).toBe(expectedTag)
+      }
+    )
+  })
+
+  describe('Encoder', () => {
+    test.each(valid)('encoding a decoded value yields original value', original => {
+      const roundtrip = pipe(
+        original,
+        BigIntString.Decoder.decode,
+        E.map(BigIntString.Encoder.encode),
+        E.getOrElseW(() => 'unexpected')
+      )
+      expect(original).toEqual(roundtrip)
+    })
+  })
+
+  describe('Eq', () => {
+    test.each(RA.zipWith(valid, valid, tuple))(
+      'determines two strings are equal',
+      (str1, str2) => {
+        const guard = BigIntString.Guard.is
+        const eq = BigIntString.Eq.equals
+        if (!guard(str1) || !guard(str2)) {
+          throw new Error('Unexpected result')
+        }
+        expect(eq(str1, str2)).toBe(true)
+      }
+    )
+  })
+
+  describe('Guard', () => {
+    test.each(cat(combineExpected(valid, true), combineExpected(invalid, false)))(
+      'validates valid strings, and catches bad strings',
+      (str, expectedTag) => {
+        const result = BigIntString.Guard.is(str)
+        expect(result).toBe(expectedTag)
+      }
+    )
+  })
+
+  describe('TaskDecoder', () => {
+    test.each(cat(combineExpected(valid, 'Right'), combineExpected(invalid, 'Left')))(
+      'validates valid string, and catches bad string',
+      async (str, expectedTag) => {
+        const result = await BigIntString.TaskDecoder.decode(str)()
+        expect(result._tag).toBe(expectedTag)
+      }
+    )
+  })
+
+  describe('Type', () => {
+    test.each(cat(combineExpected(valid, 'Right'), combineExpected(invalid, 'Left')))(
+      'validates valid strings, and catches bad strings',
+      (str, expectedTag) => {
+        const result = BigIntString.Type.decode(str)
+        expect(result._tag).toBe(expectedTag)
+      }
+    )
+  })
+
+  it('converts to bigint', () => {
+    const str = '123'
+    if (!BigIntString.Guard.is(str)) throw new Error('Unexpected result')
+    expect(BigIntString.toBigInt(str)).toEqual(BigInt(123))
+  })
```

Comment:
Tiny nit: you might've wanted to wrap this test in a `describe` block. Like, `destructors` perhaps, or `toBigInt`.

Reply:
I omitted the describe block here just because it would only be a single `it` in a whole `describe` and they would basically be the same thing